scripts/scil_make_priors_functionnectome.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Create Functionnectome priors using *.trk files from a given folder.
+Create a new folder containing all the probability maps derived from the input.
+All the tractograms should be in the same space than the brain template.
+The brain template defines which voxels are used to generate the probability map.
+"""
+
+import nibabel as nib
+import numpy as np
+import os
+import glob
+import argparse
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_output_dirs_exist_and_empty)
+
+from dipy.io.streamline import load_tractogram
+from dipy.tracking import utils
+# from dipy.tracking.streamlinespeed import length
+# from dipy.io.stateful_tractogram import StatefulTractogram
+from dipy.tracking.streamline import Streamlines
+from scilpy.tracking.tools import filter_streamlines_by_length
+# from multiprocessing import Pool  #TODO
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_dir_tractogram',
+                   help='Path of the directory containing the input tractograms (.trk).')
+    p.add_argument('in_template_file',
+                   help='Path of the brain template file (.nii, nii.gz).')
+    p.add_argument('out_dir',
+                   help='Output directory for the priors.')
+
+    # p.add_argument('--in_dir_region',  # TODO later
+    #                help='Path to the directory containing the region masks.')
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.in_dir_tractogram,
+                                 args.in_template_file,
+                                 args.in_transfo], args.in_deformation)
+    assert_output_dirs_exist_and_empty(parser, args, args.out_dir)
+
+    templ_i = nib.load(args.in_template_file).astype(bool)
+    vox_list = np.argwhere(templ_i)
+    base_mask = np.zeros(templ_i.shape, dtype=bool)
+
+    trk_list = glob.glob(os.path.join(args.in_dir_tractogram, '*.trk'))
+    for trk_F in trk_list:
+        print(os.path.basename(trk_F))
+        trk_sft = load_tractogram(trk_F, templ_i, bbox_valid_check=False)
+        _ = trk_sft.remove_invalid_streamlines()  # Ideally, only remove the sections outsite of the boundingbox
+        trk_sft = filter_streamlines_by_length(trk_sft, 25, 250)
+        trk_sft.to_vox()
+        for vox in vox_list:  # TODO Parallelize
+            vox_mask = base_mask.copy()
+            vox_mask[tuple(vox)] = True
+
+            vox_strm = trk_sft.get_streamlines_copy()
+            vox_strm = Streamlines(utils.target(vox_strm, templ_i.affine, vox_mask))
+            vox_strm = vox_strm.get_streamlines_copy()
+
+            vol_strm = np.zeros(templ_i.shape)
```

Comment:
The next few lines should be replaced by compute_tract_counts_map() in tractanalysis/streamlines_metrics

Reply:
This function computes a density map (that you can then binarize). But it is robust to large step size. Doing it by hand will skip a few voxel if you step sizes are above 0.5mm, while the cython function does a line integration.

![image](https://user-images.githubusercontent.com/10820351/133471559-97696e61-56c0-464e-a11d-cf528097df50.png)

Reply:
OK, I just tested it, no problem. Though it dosn't give the exact same voxels as my script... so I think I'll have to check how it count the streamlines

Reply:
The tractogram I'm using has a step size of 0.5mm, but voxels of 2mm in the final map, so it shouldn't be a problem. It might be due to the shift you mentioned in the other comment. I'll investigate, thanks!